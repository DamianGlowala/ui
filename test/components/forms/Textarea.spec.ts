import { describe, it, expect } from 'vitest'
import { UTextarea } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Textarea', () => {
  it.each([
    ['basic case', {}],
    ['with id', { id: 'exampleId' }],
    ['with name', { name: 'exampleName' }],
    ['with placeholder', { placeholder: 'examplePlaceholder' }],
    ['with required', { required: true }],
    ['with disabled', { disabled: true }],
    ['with rows', { rows: 5 }],
    ['with maxrows', { maxrows: 10 }],
    ['with autoresize', { autoresize: true }],
    ['with autofocus', { autofocus: true }],
    ['with autofocusDelay', { autofocusDelay: 200 }],
    ['with resize', { resize: true }],
    ['with padded', { padded: false }],
    ['with size', { size: 'sm' }],
    ['with color', { color: 'blue' }],
    ['with variant', { variant: 'outline' }],
    ['with textareaClass', { textareaClass: 'w-full' }],
    ['with modelModifiers', { modelModifiers: { trim: true, lazy: false, number: true } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Textarea.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UTextarea)
    expect(html).toMatchSnapshot()
  })
})
