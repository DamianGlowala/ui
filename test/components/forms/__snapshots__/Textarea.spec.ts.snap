// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > renders basic case correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with autofocus correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with autofocusDelay correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with autoresize correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with color correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with disabled correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with id correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with maxrows correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with modelModifiers correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with name correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with padded correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with placeholder correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with required correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with resize correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with rows correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with size correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with textareaClass correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;

exports[`Textarea > renders with variant correctly 1`] = `"<div class=""><textarea rows="3" class=""></textarea>label</div>"`;
